libsolidity/interface/CompilerStack.cpp

```diff
@@ -509,22 +509,31 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 	};
 
 	size_t longestPrefix = 0;
+	size_t longestContext = 0;
 	string longestPrefixTarget;
+
 	for (auto const& redir: m_remappings)
 	{
-		// Skip if we already have a closer match.
-		if (longestPrefix > 0 && redir.prefix.length() <= longestPrefix)
+		string contextFileString = sanitizePath(redir.context);
+		string prefixFileString = sanitizePath(redir.prefix);
+		// Skip if there is a prefix collision and the current context is closer
+		if (longestContext > 0 && contextFileString.length() < longestContext)
```

Comment:
`longestContext > 0` can be removed. If `longestContext` is zero, nothing will be smaller.